ds/dataset.go

```diff
@@ -142,11 +142,30 @@ func (d *Dataset) GetBody(thread *skylark.Thread, _ *skylark.Builtin, args skyla
 
 // SetBody assigns the dataset body
 func (d *Dataset) SetBody(thread *skylark.Thread, _ *skylark.Builtin, args skylark.Tuple, kwargs []skylark.Tuple) (skylark.Value, error) {
-	var data skylark.Iterable
-	if err := skylark.UnpackPositionalArgs("set_body", args, kwargs, 1, &data); err != nil {
+	var (
+		data skylark.Value
+		raw  skylark.Bool
+	)
+
+	if err := skylark.UnpackArgs("set_body", args, kwargs, "data", &data, "raw?", &raw); err != nil {
 		return skylark.None, err
 	}
 
+	if raw {
+		fmt.Println("raw data y'all")
+		if str, ok := data.(skylark.String); ok {
+			d.infile = cafs.NewMemfileBytes("data", []byte(string(str)))
+		} else {
+			return skylark.None, fmt.Errorf("expected raw data for body to be a string")
+		}
+		return skylark.None, nil
```

Comment:
Move this return statement to be directly after `d.inline` assignment. Get rid of `else` and unindent the error handling case by a level.